conans/model/manifest.py

```diff
@@ -7,19 +7,23 @@
 from conans.util.files import load, md5, md5sum, save, walk
 
 
-def discarded_file(filename):
+def discarded_file(filename, keep_python):
     """
     # The __conan pattern is to be prepared for the future, in case we want to manage our
     own files that shouldn't be uploaded
     """
-    return (filename == ".DS_Store" or filename.endswith(".pyc") or
-            filename.endswith(".pyo") or filename == "__pycache__" or
-            filename.startswith("__conan"))
+    if not keep_python:
+        return (filename == ".DS_Store" or filename.endswith(".pyc") or
+                filename.endswith(".pyo") or filename == "__pycache__" or
+                filename.startswith("__conan"))
+    else:
+        return filename == ".DS_Store"
```

Comment:
Add here the `filename.startswith("__conan")`, I think that the motivation was to reserve a filename just in case we want to add something in that folder in the future.